src/connector/builder.rs

```diff
@@ -174,16 +174,22 @@ impl ConnectorBuilder<WantsProtocols1> {
         })
     }
 
-    /// Enable all HTTP versions
+    /// Enable all HTTP versions built into this library
     ///
-    /// For now, this enables both HTTP 1 and 2. In the future, other supported versions
-    /// will be enabled as well.
-    #[cfg(all(feature = "http1", feature = "http2"))]
+    /// For now, this could enable both HTTP 1 and 2, depending on active features.
+    /// In the future, other supported versions will be enabled as well.
+    #[cfg(feature = "http2")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "http2")))]
```

Comment:
If we're going to do this, let's get rid of the `http2` guard? And make it work for all of the possible combinations?

Reply:
Agreed, that seems like the right direction

Reply:
@djc

The problem with not having the `http2` guard is that the return type would change (since `WantsProtocols3` wont exist). So if any library enables the feature it might cause other libraries to not compile if they expected `WantsProtocols2`. Since enabling an extra feature in a deep dependency doesn't usually warrant a major version release, things in the wild could spontaneously break if one single library enables it when one other doesn't expect it.

Making the feature combination non-breaking would require wrapping the actual inner type `WantsProtocolsN` (based on the highest enabled protocol feature) in a `WantsProtocolsAll` singleton struct and always returning that. That would be a breaking change for this library, however, which is why I didn't want to do it in this fix.

If a version of this fix gets merged I could create a new PR with the above solution with a new breaking return type.

Reply:
That seems reasonable, I'd be willing to merge this in order to get this improvement out provided you submit an additional PR to do the semver-breaking fix.